### ISSUE TITLE
Promote js_image_layer to Public API

### ISSUE THREAD

**Issue by thesayyn** [aspect-build/rules_js#304]
js_image_layer should be promoted to public API before 1.0

**Comment by gregmagolan** [comment]
Not landing this for 1.0 since there are transitive deps on rules_py and rules_pkg that we have to reason about first

**Comment by bazaglia** [comment]
As spoken with @thesayyn on Slack, I tested the `js_image_layer` following the example on the e2e folder and noticed something additional had to be done when importing files. It can be either improved or be part of the docs.

When running the image built using `js_image_layer`, my Node.js runtime at first wasn't able to find some `.proto` files that I import. It all works great when using `js_binary` directly though. But to get the files to be found with this Starlark code I'm importing from the e2e folder, I had to:

* Set the `workdir` attribute in the `container_image` to `/app/my_package/start.sh.runfiles/my_workspace_name`  (`/app`  is the root in `js_image_layer`). It also works if I set the environment variable ` JS_BINARY__CHDIR` to that same string.
* After setting the workdir as mentioned above, I still need to prefix the file I import with `process.cwd()`. When I call the `js_binary` target directly without Docker, I don't need to prefix it.
